### PR TITLE
Add batch_norm function to paddle frontend 

### LINKED ISSUE #17377: batch_norm

### PR DESCRIPTION
**Pull request by melch-inno**
close #17377

### CODE REVIEW

**Comment by danielmunioz** [comment]
Hey @melch-inno!, Thanks for contributing!

It looks like you have some misnamed and some missing arguments in your function, could you please take a look at those?, a frontend implementation should resemble its [original counterpart](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/nn/functional/batch_norm_en.html) as closely as possible

Also it looks like you added a new file named `run_tests_CLI/test_paddle_frontend.sh` could you please remove that as well, the PR should ideally only contain modifications relevant to the issue at hand :)

Thanks!!

**Comment by AdityaSaxena2121** [comment]
-[ ] #17413

**Comment by danielmunioz** [comment]
Hey @melch-inno, so it seems that you are using `eps` instead of `epsilon` in the frontend testing helper and so the function cannot properly recognize the argument name, argument names in the helper should match the ones in the frontend function so that it can work properly (checkout the [frontend tests guide](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html#ivy-frontend-tests) for more info).

Also the `out` argument of `ivy.batch_norm` is just an optional for output arrays, for writing the result to, paddle batch_norm doesn't seem to utilize it so there is no need to pass it into the ivy function.

Same goes for `name` and `use_global_stats`, these two are not directly being used in the function, while we need to keep them as arguments so as to not break frontend compatibility we can remove them from the function itself, check the [unused arguments](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#unused-arguments) section of the deep dive, as well as the [frontend guide](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.htm) itself which contains useful information about frontend functions.

If you have any questions please feel free to reach out!
Thanks

**Comment by melch-inno** [comment]
Hi @danielmunioz , updates pushed. Pls, check when you are available. thanks~

**Comment by melch-inno** [comment]
Hi @theRealBird , Pls kindly check this PR when you are available. Thanks ~

**Comment by danielmunioz** [comment]
Hey @melch-inno! Could you please update the supported version to 2.5.0? and update the dtypes accordingly if they need to? After that could you please run the tests and make sure they're passing?

Thanks!

File: ivy/functional/backends/paddle/experimental/norms.py [hunk 1]
```diff
@@ -28,8 +28,8 @@
 )
 def batch_norm(
     x: paddle.Tensor,
-    mean: paddle.Tensor,
-    variance: paddle.Tensor,
+    running_mean: paddle.Tensor,
```
Comment (melch-inno) [inline ivy/functional/backends/paddle/experimental/norms.py:31]: `Positional keywords mean, variance is not allowed in Ivy to be passed as  **args`.  making changes to align with the original paddle function args.

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py [hunk 2]
```diff
@@ -190,8 +190,8 @@ def test_batch_norm(
         atol_=1e-2,
         input_dtypes=x_dtype,
         x=x,
-        mean=mean,
-        variance=variance,
+        running_mean=mean,
```
Comment (melch-inno) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_norms.py:193]: updating test to reflect changes

**Comment by danielmunioz** [comment]
Hey! @melch-inno, Sorry but your recent changes seem to be altering the backends, the backend implementations don't follow the structure of the original frameworks, they're more focused on functionality and unifying the behavior of the functions across all backends, so updating them to look like their counterparts is not necessary

Could you please remove the changes made in the backends? (Also, as per ivy rules, a PR needs to only contain modifications related to the issue at hand, so for this case anything other than changes related to paddle_front batch_norm should go in another PR)

Please feel free to ask away!
Thanks! :)

**Comment by melch-inno** [comment]
> Hey! @melch-inno, Sorry but your recent changes seem to be altering the backends, the backend implementations don't follow the structure of the original frameworks, they're more focused on functionality and unifying the behavior of the functions across all backends, so updating them to look like their counterparts is not necessary
> 
> Could you please remove the changes made in the backends? (Also, as per ivy rules, a PR needs to only contain modifications related to the issue at hand, so for example, in this case, anything other than changes related to paddle_front batch_norm should go into another PR)
> 
> Please feel free to ask away! Thanks! :)



Hi @danielmunioz, I understand, but we might have to create a new ticket/task to make those changes at the backend because the paddle backend does not allow positional-only arguments passed as keyword arguments. This PR can be put on hold for those changes to be done before I can continue. see the exception below for reference; thanks ~

Screenshot 2023-07-13 at 1 32 18 PM

**Comment by danielmunioz** [comment]
I'll take a look at those backend errors, but right now there seems to still be a merge error as well as an unexpected argument in the tests, seems that the `y` arguments is unused there, could you please fix those?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by melch-inno** [comment]
@danielmunioz , updates pushed

**Comment by BilkisAhmed** [comment]
@melch-inno can I work on this issues

**Comment by danielmunioz** [comment]
Hey @melch-inno! Could you please remove the usage of the `paddle` package? the frontend is meant to be used using frontend arrays, so casting to paddle arrays is not necessary!

**Comment by danielmunioz** [comment]
Hey @BilkisAhmed! This task is currently a work in progress, but if you'd like to contribute to ivy please check out the [Open Tasks](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html) that we have available!

**Comment by danielmunioz** [comment]
Closing due to inactivity

**Pull request closed without merging**